### PR TITLE
feat: Master quiz , modal for additional info 

### PR DESCRIPTION
**Pull request by firassssss**
## Fixes Issue

https://github.com/thecyberworld/TheCyberHUB/issues/632

## Changes proposed

adding a button next to the selected wrong question , when cliking on it , it opens a modal that contains 
additional info about the question 

## Screenshots

Capture d'écran 2024-04-27 112723
Capture d'écran 2024-04-27 115545

## Note to reviewers

the additional info should be added in the CategoriesData.jsx under additionalInfo 

## Code of Conduct

- [ ] By submitting this pull request, I confirm I've read and complied with the [CoC](https://github.com/thecyberworld/TheCyberHUB/blob/dev/CODE_OF_CONDUCT.md) 🖖

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

---

You can also join our [Discord](https://discord.gg/QHBPq6xP5p) community.
Feel free to check out other cool repositories of the [Thecyberworld](https://github.com/thecyberworld).
Join the Thecyberworld GitHub Organisation by raising an [issue](https://github.com/thecyberworld/Support/issues/new?assignees=&labels=invite+me+to+the+organisation&template=invitation.yml&title=Please+invite+me+to+the+GitHub+Community+Organization) (you will be sent an invitation).

### CODE REVIEW

File: src/components/Resources/Quiz/Categories/QuizPage.jsx [hunk 1]
```diff
@@ -150,30 +168,79 @@ export default function QuizPage() {
...
+                                                    borderRadius: "5px",
+                                                }}
+                                            >
+                                                afezfbfibzfbzbfezbfbezzfbfbijbbfzfff iueb f f hfen c v v e vbe vebvv
+                                                nnffbfefbjfnefefnejnfjenff fn fnuf i jafinevjnzkfdiezjfnz vj ejf vef ee
+                                                vfeunejiuir rencnerc freuer vuire reinjznjzeof zef efiiv rjr f ervne f
+                                                nvevf vfenv rf revevne fn reinvev evue vebve jveiuv h
+                                                jnrjfnerjnjnijnegene vnv e vri jner v jnf erfref v feij ionz ezf &jv vi
+                                                invenvvjev vnvevev iefenrbvnjre e r en n iuznfjnerzfv nn n v{" "}
+                                                {questions[currentQuestion].additionalInfo}
```
Comment (ArkadiK94) [inline src/components/Resources/Quiz/Categories/QuizPage.jsx:195]:
```suggestion
                                                {questions[currentQuestion].additionalInfo}
```
for now it is enough like this :)

File: src/pages/AuthPopup/AuthPopupElements.jsx [hunk 2]
```diff
@@ -41,8 +41,8 @@ export const PopupContainer = styled.div`
 
 export const CloseButton = styled(ImCross)`
     position: absolute;
-    top: 20px;
-    right: 20px;
+    top: 6px;
+    right: 6px;
```
Comment (ArkadiK94) [inline src/pages/AuthPopup/AuthPopupElements.jsx:45]: before:
image

after:
image

FeedPagePost.jsx , it looks there too close to the border, I like the previous option. what do you think?

File: src/components/Resources/Quiz/Categories/QuizPage.jsx [hunk 3]
```diff
@@ -37,6 +40,7 @@ export default function QuizPage() {
     const [clickedAnswerIndex, setClickedAnswerIndex] = useState(null);
     const [buttonClicked, setButtonClicked] = useState(false);
     const [disableSkipButton, setDisableSkipButton] = useState(true);
+    const [ShowAdditionalInfo, setShowAdditionalInfo] = useState(false);
```
Comment (ArkadiK94) [inline src/components/Resources/Quiz/Categories/QuizPage.jsx:43]:
```suggestion
    const [showAdditionalInfo, setShowAdditionalInfo] = useState(false);
```
also in all places where you used this variable

File: src/components/Resources/Quiz/Categories/QuizPage.jsx [hunk 4]
```diff
@@ -89,6 +93,9 @@ export default function QuizPage() {
             setShowAnswer(true);
             setClickedAnswerIndex(i);
             setButtonClicked(true);
+            if (ShowAdditionalInfo) {
```
Comment (ArkadiK94) [inline src/components/Resources/Quiz/Categories/QuizPage.jsx:96]:
```suggestion
            if (showAdditionalInfo) {
```

File: src/components/Resources/Quiz/Categories/QuizPage.jsx [hunk 5]
```diff
@@ -150,30 +168,79 @@ export default function QuizPage() {
...
-                                                : "",
-                                    }}
-                                    disabled={buttonClicked}
-                                >
-                                    {answerOption.answerText}
-                                </QuestionButton>
+                                <div key={i} style={{ position: "relative", zIndex: 0 }}>
+                                    {showAnswer && !answerOption.isCorrect && i === clickedAnswerIndex && (
+                                        <InfoButton onClick={(event) => handleAdditionalInfoButton(event)}>
+                                            <BsInfoCircle />
```
Comment (ArkadiK94) [inline src/components/Resources/Quiz/Categories/QuizPage.jsx:224]: on hover add tooltip - few words near to the icon, so the user will know on what they click, and that they should click it, or change to different icon which will indicate it better.

**Review by ArkadiK94 — CHANGES_REQUESTED** [review]
Thank you very much for your contribution :)
Looks great but I have some comments.


https://github.com/thecyberworld/TheCyberHUB/assets/76536506/0c0defde-5df4-44ec-b910-b5b41f37795b

In addition to the other comments, 
First, I think this icon should be near the correct answer. Second, when the user clicks on correct answer it doesn't mean that they will not want to see more insightful info related to this answer especially because sometimes the user can guess the answer if they don't know it.
For this reason, the same functionality should apply also for a case when the picked answer is correct.

**Comment by firassssss** [comment]
i will work on resolving them

File: src/pages/AuthPopup/AuthPopupElements.jsx (hunk 2 above)
Comment (Salem-Bourguiba) [reply src/pages/AuthPopup/AuthPopupElements.jsx:45]: yes you are right it is surely too close , but the first one is too far , i think in the middle will be perfect

Comment (firassssss) [reply src/pages/AuthPopup/AuthPopupElements.jsx:45]: yes you are right i think it is too close but the first one is too far , i think in the middle will be perfect

File: src/components/Resources/Quiz/Categories/QuizPage.jsx (hunk 3 above)
Comment (firassssss) [reply src/components/Resources/Quiz/Categories/QuizPage.jsx:43]: i did not understand your question

**Comment by firassssss** [comment]
> Thank you very much for your contribution :) Looks great but I have some comments.
> 
>  Recording.2024-04-27.142516.mp4 
> In addition to the other comments, First, I think this icon should be near the correct answer. Second, when the user clicks on correct answer it doesn't mean that they will not want to see more insightful info related to this answer especially because sometimes the user can guess the answer if they don't know it. For this reason, the same functionality should apply also for a case when the picked answer is correct.

but i the right answer is selected it will automatically skip to the next question after 2seconds , are preposing to remove this functionality ?

File: src/components/Resources/Quiz/Categories/QuizPage.jsx (hunk 3 above)
Comment (ArkadiK94) [reply src/components/Resources/Quiz/Categories/QuizPage.jsx:43]: I don't have question, change to lower case in all places that you use is variable.

**Comment by ArkadiK94** [comment]
@firassssss yes

File: src/components/Resources/Quiz/Categories/QuizPage.jsx (hunk 3 above)
Comment (firassssss) [reply src/components/Resources/Quiz/Categories/QuizPage.jsx:43]: aa ok sorry

**Comment by firassssss** [comment]
@ArkadiK94,  @Salem-Bourguiba is my colleague and we are collaborating to finish this issue

**Comment by ArkadiK94** [comment]
@firassssss Hi, I saw that you made some changes, is it ready for another review or is it still in progress?

**Comment by firassssss** [comment]
> @firassssss Hi, I saw that you made some changes, is it ready for another review or is it still in progress?
it is ready for another review

File: src/components/Resources/Quiz/Categories/CategoriesElements.jsx [hunk 6]
```diff
@@ -61,10 +61,62 @@ export const SkipButton = styled.button`
...
+`;
+export const TooltipText = styled.div`
+    position: relative;
+    top: 27px;
+    right: 89px;
+    display: flex;
+    align-items: center;
+    font-weight: 100;
+    width: 58px;
+    background: ;
```
Comment (ArkadiK94) [inline src/components/Resources/Quiz/Categories/CategoriesElements.jsx:117]:
```suggestion

```

File: src/components/Resources/Quiz/Categories/QuizPage.jsx [hunk 7]
```diff
@@ -150,30 +163,80 @@ export default function QuizPage() {
...
+                                    {showAnswer && answerOption.isCorrect && (
+                                        <div style={{ height: "0px" }}>
+                                            <InfoButton
+                                                onClick={(event) => handleAdditionalInfoButton(event)}
+                                                onMouseEnter={handleMouseEnter}
+                                                onMouseLeave={handleMouseLeave}
+                                            >
+                                                <BsInfoCircle />
+                                            </InfoButton>
+                                            {showTooltip && <TooltipText>additional information</TooltipText>}
```
Comment (ArkadiK94) [inline src/components/Resources/Quiz/Categories/QuizPage.jsx:220]:
```suggestion
                                            {showTooltip && <TooltipText>Click here for additional information</TooltipText>}
```

**Review by ArkadiK94 — CHANGES_REQUESTED** [review]
@firassssss  Thanks for your contribution looks much better.
Only two minor adjustments.

**Review by ArkadiK94 — APPROVED** [review]
@firassssss Thanks for your contribution 💯

**Pull request merged**